src/backend/decayrange.h

```diff
@@ -0,0 +1,276 @@
+#pragma once
+
+#include "../ds/ptrwrap.h"
+#include "largebuddyrange.h"
+#include "../pal/pal_ds.h"
+
+namespace snmalloc
+{
+  template<typename Rep>
```

Comment:
Given the reuse of the large buddy range rep here, at least a comment (or a `concept`) might be in order.